### ISSUE TITLE
500 / 404 - Cluster: there is no endpoint but we've a hash

### ISSUE THREAD

**Issue by olivierpascal** [meteorhacks/cluster#72]
I remove all db.clusterEndpoints documents, delete all browser cookies and restart the server. Then:

https://mydomain.com/cluster-ddp/6d18937c744cb670d464d45759837ff16eb232a7/web/694/p387anjk/xhr Failed to load resource: the server responded with a status of 500 (Internal Server Error)

POST https://a1.mydomain.com/cluster-ddp/6d18937c744cb670d464d45759837ff16eb232a7/web/216/syax6nda/xhr
XMLHttpRequest cannot load https://a1.mydomain.com/cluster-ddp/6d18937c744cb670d464d45759837ff16eb232a7/web/216/syax6nda/xhr. No 'Access-Control-Allow-Origin' header is present on the requested resource. Origin 'https://mydomain.com' is therefore not allowed access. The response had HTTP status code 404.

In logs:

```
Cluster: there is no endpoint but we've a hash: 6d18937c744cb670d464d45759837ff16eb232a7
```

My envvars:

```
ROOT_URL="https://mydomain.com";
CLUSTER_SERVICE="web";
CLUSTER_ENDPOINT_URL="https://a1.mydomain.com";
CLUSTER_BALANCER_URL="https://a1.mydomain.com";
CLUSTER_WORKERS_COUNT="1";
DISABLE_WEBSOCKETS="1";
```

The DB:

```
db.clusterEndpoints.find().pretty()
{
	"_id" : ObjectId("55a296f86924317b7a4b2fef"),
	"endpoint" : "https://a1.mydomain.com",
	"serviceName" : "web",
	"timestamp" : ISODate("2015-07-12T16:41:42.220Z"),
	"pingInterval" : 5000,
	"endpointHash" : "6d18937c744cb670d464d45759837ff16eb232a7",
	"balancer" : "https://a1.mydomain.com"
}
```

**Comment by olivierpascal** [comment]
Got it!! Guess what? The service name cannot contains special chars like dot, underscore, dash, etc. My service name was "projectname-web". It seems that the route cannot map this kind of name and always return a 404. Waaah. I can move on!

**Comment by olivierpascal** [comment]
Correction: the underscore is working. My service name is now "project_web".

**Comment by petermikitsh** [comment]
This is a gotcha! Experienced this myself.

**Comment by pmuens** [comment]
Thank you so much! This resolved my issue #94 as well!
